code/workspaces/web-app/src/api/stackService.js

```diff
@@ -43,7 +43,12 @@ function getUrl(projectKey, id) {
     });
 }
 
-function createStack(stack) {
+function createStack(stackWithAssets) {
+  const stack = {
+    ...stackWithAssets,
+    assetIds: stackWithAssets.assets ? stackWithAssets.assets.map(asset => asset.assetId) : [],
+  };
+  delete stack.assets;
```

Comment:
Are the stack assets being removed a temporary measure until the back-end is in place to handle them? If that is the case, I don't think this needs to be done as the infrastructure api should ignore the field until it is specified in the validators on the infrastructure api endpoints.

Reply:
No, it's not a temporary measure; rather than passing stack.assets (which would therefore have extraneous information such as the asset names and file locations), we're only passing assetIds.  If we leave them in, graphql will complain because they're not part of StackCreationRequest.

Reply:
Of course, sorry, I had missed the fact that it was a new field `assetIds` being created!

At what point would the stacks get the assets array assigned to it? And does it need to have it? Would it be possible to just use an array of IDs without having the intermediary array of assets?

Reply:
The stack that gets here is basically as it appears on the form, which is why it has additional fields apart from assetId.

Reply:
Ahhh I see! Thank you for explaining Andy. In that case I am happy for this to go!